launch_ros/launch_ros/actions/lifecycle_transition.py

```diff
@@ -0,0 +1,257 @@
+        Transition.TRANSITION_CLEANUP:
+            {'start_state': 'cleaningup', 'goal_state': 'unconfigured'},
+        Transition.TRANSITION_ACTIVATE:
+            {'start_state': 'activating', 'goal_state': 'active'},
+        Transition.TRANSITION_DEACTIVATE:
+            {'start_state': 'deactivating', 'goal_state': 'inactive'},
+        Transition.TRANSITION_UNCONFIGURED_SHUTDOWN:
+            {'start_state': 'shuttingdown', 'goal_state': 'finalized'},
+        Transition.TRANSITION_INACTIVE_SHUTDOWN:
+            {'start_state': 'shuttingdown', 'goal_state': 'finalized'},
+        Transition.TRANSITION_ACTIVE_SHUTDOWN:
+            {'start_state': 'shuttingdown', 'goal_state': 'finalized'},
+    }
+
+    def __init__(
+            self,
+            *,
+            lifecycle_node_names: Iterable[SomeSubstitutionsType],
+            transitions_ids: Iterable[Union[int, SomeSubstitutionsType]],
+            **kwargs, ) -> None:
+        """
+        Construct a LifecycleTransition action.
+
+        The action will execute the passed in lifecycle transition for the
+        lifecycle nodes with the indicated node names. The action will emit
+        an event that triggers the first lifecycle transition of each node
+        wait that the node reaches the transition goal and trigger the next
+        transition in the list.
+        You need to make sure, that the sequence of lifecyle transition you
+        pass in is possible.
+
+        :param lifecycle_node_names: The names of the lifecycle nodes to transition
+        :param transitions_ids: The transitions to be executed.
+        """
+        super().__init__(**kwargs)
+        self.__lifecycle_node_names = [
+            normalize_to_list_of_substitutions(name)
+            for name in lifecycle_node_names]
+        temp_transitions_ids = []
+        for id in transitions_ids:
+            temp_id = None
+            if isinstance(id, int):
+                temp_id = str(id)
+            else:
+                temp_id = id
+            temp_transitions_ids.append(temp_id)
+
+        self.__transition_ids = [
+            normalize_to_list_of_substitutions(id)
+            for id in temp_transitions_ids]
+
+        self.__event_handlers = {}
+        self.__logger = launch.logging.get_logger(__name__)
+
+    @classmethod
+    def parse(cls, entity: Entity, parser: Parser):
+        """Parse load_composable_node."""
+        _, kwargs = super().parse(entity, parser)
+
+        kwargs['lifecycle_node_names'] = parser.parser_substitution(
+            entity.get_attr('lifecycle_node_names', data_type=List[str])
+        )
+        # Probably need to make this parsable through a dict for convenience
+        kwargs['transitions_ids'] = entity.get_attr(
+            'transitions_ids', data_type=List[int])
+        return cls, kwargs
+
+    def _remove_event_handlers(
+            self,
+            context: LaunchContext,
+            node_name: str = None,
```

Comment:
It doesn't look like we ever call this without a node name
```suggestion
            node_name: str
```